pages/nft/url/index.vue

```diff
@@ -392,56 +424,60 @@ export default class FetchIndex extends Vue {
 
   async doAction() {
     this.errorMessage = ''
+    this.hasError = false
     /* eslint-disable no-fallthrough */
-    switch (this.state) {
-      case State.INIT: {
-        if (this.ownerWallet && this.address !== this.ownerWallet) {
-          this.errorMessage = 'PLEASE_USE_OWNER_WALLET_TO_SIGN'
-          break
-        }
-        if (this.iscnPrefixRegex.test(this.url)) {
-          this.iscnId = this.url;
-          this.$router.push(
-            this.localeLocation({
-              name: 'nft-iscn-iscnId',
-              params: this.iscnParams,
-              query: this.queryParams,
-            })!,
-          )
-          break
-        }
-
-        this.balance = await getAccountBalance(this.address) as string
-        if (this.balance === '0') {
-          this.toggleSnackbar('INSUFFICIENT_BALANCE')
-          break
+    try {
+      switch (this.state) {
+        case State.INIT: {
+          if (this.ownerWallet && this.address !== this.ownerWallet) {
+            throw new Error('PLEASE_USE_OWNER_WALLET_TO_SIGN')
+          }
+          if (this.iscnPrefixRegex.test(this.url)) {
+            this.iscnId = this.url
+            this.$router.push(
+              this.localeLocation({
+                name: 'nft-iscn-iscnId',
+                params: this.iscnParams,
+                query: this.queryParams,
+              })!,
+            )
+            break
+          }
+
+          this.balance = (await getAccountBalance(this.address)) as string
+          if (this.balance === '0') {
+            throw new Error('INSUFFICIENT_BALANCE')
+          }
+
+          this.$router.replace({
+            query: {
+              ...this.$route.query,
+              url: this.url,
+            },
+          })
+          this.state = State.TO_CRAWL_URL
         }
-
-        this.errorMessage = ''
-        if (!this.url) {
-          this.errorMessage = this.$t('HomePage.search.errormessage.empty') as string
+        case State.TO_CRAWL_URL:
+          await this.crawlUrlData()
+          this.state = State.TO_ESTIMATE_ARWEAVE_FEE
+        case State.TO_ESTIMATE_ARWEAVE_FEE:
+          await this.checkArweaveIdExistsAndEstimateFee()
+          this.state = this.arweaveId
+            ? State.TO_REGISTER_ISCN
+            : State.TO_UPLOAD_TO_ARWEAVE
+        case State.TO_UPLOAD_TO_ARWEAVE:
+          if (!this.arweaveFeeTxHash) {
+            await this.sendArweaveFeeTx(this.arweaveFeeInfo)
+          }
+          await this.submitToArweave()
+          this.state = State.TO_REGISTER_ISCN
+        case State.TO_REGISTER_ISCN:
+          await this.registerISCN()
+        default:
           break
-        }
```

Comment:
If the input doesn't have a URL, the button will be disabled and can't proceed to the next step, so I think this can be removed. Or do you think I should keep it?

Reply:
Sure let's remove it